### DISCUSSION TITLE
[Elasticsearch] Field selection lost when moving between any Metrics/Bucket with fields.

### DISCUSSION THREAD

**Discussion by Ultraseamus** [grafana/grafana#61171] [Data Sources]
**What happened**:

Created an Elasticsearch query with a metric aggregation of "Min: @value".
Switched "Min" to "Max".
"@value" was replaced with the default no selection.

**What you expected to happen**:
As with previous versions of Grafana, it should remember your field selection.

**Anything else we need to know?**:
This also applies to every aggregation with shared options.

Looking at the source code, I think this change may have been intentional. But in practice, it is a worse experience. When you switch between similar aggregation types, it is reasonable to expect the options that those types share to stick around.

It is common for users to select the field they are interested in first, and then decide which aggregation type they want. This change makes that experience worse.

My guess is that the change was made so that unused options were cleaned up as queries were edited. Which is a good cause, but there needs to be an exception when you are changing between aggregations with shared options (especially "Field", but also all of the other options).

**Environment**:
- Grafana version: 8.0.6
- Data source type & version: Elasticsearch

**Comment by matyax** [comment]
We need to follow up on this one.

**Comment by gelicia** [comment]
Hello, as you may have heard, we are [transitioning away from using discussions to discuss feature requests](https://github.com/grafana/grafana/discussions/73424). Due to the age and number of responses to this discussion, we are deciding to close it. If this is something you would like to see in Grafana, feel free to open an [issue](https://github.com/grafana/grafana/issues) so the discussion can continue. Thank you!
